packages/backend/src/peripherals/database/PreprocessedUserStatisticsRepository.ts

```diff
@@ -50,7 +56,47 @@ export class PreprocessedUserStatisticsRepository extends BaseRepository {
       .orderBy('state_update_id', 'desc')
       .first()
 
-    if (row) return toPreprocessedUserStatisticsRecord(row)
+    return row ? toPreprocessedUserStatisticsRecord(row) : undefined
+  }
+
+  async findLastWithL2TransactionsStatisticsByStarkKey(
+    starkKey: StarkKey,
+    trx?: Knex.Transaction
+  ) {
+    const knex = await this.knex(trx)
+    const row = await knex('preprocessed_user_statistics')
+      .whereNotNull('l2_transactions_statistics')
+      .andWhere('stark_key', starkKey.toString())
+      .orderBy('state_update_id', 'desc')
+      .first()
+
+    return row ? toPreprocessedUserStatisticsRecord(row) : undefined
+  }
+
+  async getAllWithoutL2TransactionStatisticsUpToStateUpdateId(
```

Comment:
Check my other comments, but this function probably is not necessary at all, because we can't depend on state update statistics for L2 transaction preprocessing, because there are scenarios when user had no state update but he has transactions (i.e. -1 and then +1)